src/features/rewards/mediaBox/index.tsx

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import {
+  StyledMediaBox,
+  StyledMediaText,
+  StyledMediaTimestamp,
+  StyledMediaIcon
+} from './style'
+
+import {
+  TwitterColorIcon,
+  RedditColorIcon
+} from '../../../components/icons'
+
+export interface Props {
+  mediaType: string
+  mediaText: string
+  mediaTimestamp: number
+  mediaTimetext?: string
+}
+
+export default class MediaBox extends React.Component<Props, {}> {
+  formatTimestamp = (timestamp: Date) => {
+    const dateOptions = { month: 'short', day: 'numeric' }
+    if (new Date().getFullYear() !== timestamp.getFullYear()) {
+      (dateOptions as any)['year'] = 'numeric'
+    }
+    return timestamp.toLocaleString(navigator.language, dateOptions)
+  }
+
+  render () {
+    return (
+      <StyledMediaBox>
+        <StyledMediaIcon>
+        {
+          this.props.mediaType === 'twitter'
+          ? <TwitterColorIcon />
+          : this.props.mediaType === 'reddit'
+          ? <RedditColorIcon />
+          : null
+        }
+        </StyledMediaIcon>
+        <StyledMediaTimestamp>
+          {
```

Comment:
could you extract this into separate function, so that render is small

Reply:
Done